### PR TITLE
doc: Rename stable API change to breaking change

### PR DESCRIPTION
**Pull request by carlescufi**
It is often confusing for users and developers alike to see the sentence "stable API change" in a label or in the release notes. Stable APIs can change in at least two ways (retaining compatibility or not), and so it is preferrable to use a term that clearly describes the change as incompatible, by using the common term "breaking".

Note: I will rename the label once this is merged.

### CODE REVIEW

File: doc/develop/api/api_lifecycle.rst [hunk 1]
```diff
@@ -94,20 +94,20 @@ In order to declare an API ``stable``, the following steps need to be followed:
...
+Introducing breaking changes
+============================
 
 A stable API, as described above strives to remain backwards-compatible through
 its life-cycle. There are however cases where fulfilling this objective prevents
 technical progress or is simply unfeasible without unreasonable burden on the
 maintenance of the API and its implementation(s).
 
-An incompatible change is defined as one that forces users to modify their
+An breaking change is defined as one that forces users to modify their
```
Comment (nashif) [inline doc/develop/api/api_lifecycle.rst:107]: A breaking....

File: doc/develop/api/api_lifecycle.rst [hunk 2]
```diff
@@ -94,20 +94,20 @@ In order to declare an API ``stable``, the following steps need to be followed:
...
 existing code in order to maintain the current behavior of their application.
 The need for recompilation of applications (without changing the application
-itself) is not considered an incompatible change.
+itself) is not considered an breaking change.
```
Comment (nashif) [inline doc/develop/api/api_lifecycle.rst:110]: a breaking

**Review by nashif — CHANGES_REQUESTED** [review]

**Review by nashif — DISMISSED** [review]

File: doc/develop/api/api_lifecycle.rst [hunk 3]
```diff
@@ -94,20 +94,20 @@ In order to declare an API ``stable``, the following steps need to be followed:
    `Zephyr Architecture meeting`_ where, barring any objections, the Pull Request
    will be merged
 
-.. _stable_api_changes:
+.. _breaking_api_changes:
 
-Introducing incompatible changes
-================================
+Introducing breaking changes
```
Comment (kartben) [inline doc/develop/api/api_lifecycle.rst:99]: For consistency and extra clarity, I would recommend to use "breaking **API** change(s)" everywhere.

```suggestion
Introducing breaking API changes
```

File: doc/develop/api/api_lifecycle.rst [hunk 4]
```diff
@@ -94,20 +94,20 @@ In order to declare an API ``stable``, the following steps need to be followed:
...
+============================
 
 A stable API, as described above strives to remain backwards-compatible through
 its life-cycle. There are however cases where fulfilling this objective prevents
 technical progress or is simply unfeasible without unreasonable burden on the
 maintenance of the API and its implementation(s).
 
-An incompatible change is defined as one that forces users to modify their
+A breaking change is defined as one that forces users to modify their
```
Comment (kartben) [inline doc/develop/api/api_lifecycle.rst:107]:
```suggestion
A breaking API change is defined as one that forces users to modify their
```

File: doc/develop/api/api_lifecycle.rst [hunk 5]
```diff
@@ -94,20 +94,20 @@ In order to declare an API ``stable``, the following steps need to be followed:
...
 existing code in order to maintain the current behavior of their application.
 The need for recompilation of applications (without changing the application
-itself) is not considered an incompatible change.
+itself) is not considered a breaking change.
```
Comment (kartben) [inline doc/develop/api/api_lifecycle.rst:110]:
```suggestion
itself) is not considered a breaking API change.
```

File: doc/develop/api/api_lifecycle.rst [hunk 6]
```diff
@@ -177,8 +177,7 @@ If the Pull Request is merged then an email must be sent to the ``devel`` and
 
 .. note::
 
-   Incompatible changes will be announced in the "API Changes" section of the
-   release notes.
+   Breaking changes will be listed and described in the migration guide.
```
Comment (kartben) [inline doc/develop/api/api_lifecycle.rst:180]:
```suggestion
   Breaking API changes will be listed and described in the migration guide.
```

File: doc/develop/api/api_lifecycle.rst [hunk 7]
```diff
@@ -94,20 +94,20 @@ In order to declare an API ``stable``, the following steps need to be followed:
...
 
-.. _stable_api_changes:
+.. _breaking_api_changes:
 
-Introducing incompatible changes
-================================
+Introducing breaking changes
+============================
 
 A stable API, as described above strives to remain backwards-compatible through
```
Comment (kartben) [inline doc/develop/api/api_lifecycle.rst:102]: Very much a nit as this is not related with the initial scope of this PR at all, but if you feel like also addressing some minor grammar tweaks (there might be other missing commas, just had a quick look, feels like it would be worth doing a full pass).

```suggestion
A stable API, as described above, strives to remain backwards-compatible through
```

File: doc/develop/api/api_lifecycle.rst [hunk 8]
```diff
@@ -94,20 +94,20 @@ In order to declare an API ``stable``, the following steps need to be followed:
...
 its life-cycle. There are however cases where fulfilling this objective prevents
 technical progress or is simply unfeasible without unreasonable burden on the
```
Comment (kartben) [inline doc/develop/api/api_lifecycle.rst:104]:
```suggestion
technical progress, or is simply unfeasible without unreasonable burden on the
```

File: doc/develop/api/api_lifecycle.rst [hunk 9]
```diff
@@ -94,20 +94,20 @@ In order to declare an API ``stable``, the following steps need to be followed:
...
 
-An incompatible change is defined as one that forces users to modify their
+A breaking change is defined as one that forces users to modify their
 existing code in order to maintain the current behavior of their application.
 The need for recompilation of applications (without changing the application
-itself) is not considered an incompatible change.
+itself) is not considered a breaking change.
 
 In order to restrict and control the introduction of a change that breaks the
 promise of backwards compatibility the following steps must be followed whenever
```
Comment (kartben) [inline doc/develop/api/api_lifecycle.rst:113]:
```suggestion
promise of backwards compatibility, the following steps must be followed whenever
```

**Review by kartben — CHANGES_REQUESTED** [review]

**Review by kartben — DISMISSED** [review]

**Comment by carlescufi** [comment]
@kartben had to push to fix a minor doc warning.

**Review by kartben — APPROVED** [review]
Ah yes, sorry I typically try to provide "full" requested changes but forgot this time around :)

**Review by nashif — APPROVED** [review]
